src/main/java/org/broadinstitute/hellbender/tools/spark/sv/StructuralVariantContext.java

```diff
@@ -0,0 +1,280 @@
+     * undefined contig names (.)
+     *
+     * @return never {@code null}, an empty list if no structural variant is specified.
+     */
+    public List<String> getContigNames() {
+        if (!hasAttribute(GATKSVVCFConstants.CONTIG_NAMES)) {
+            return Collections.emptyList();
+        } else {
+            final List<String> contigNames = getAttributeAsStringList(GATKSVVCFConstants.CONTIG_NAMES, null);
+            if (contigNames.contains(null)) {
+                throw new IllegalStateException("the contig names annotation contains undefined values");
+            }
+            return contigNames;
+        }
+    }
+
+    /**
+     * Composes the haplotype that corresponds to an allele based on the reference sequence.
+     * <p>
+     *     The resulting haplotype will be annotated with the appropriate Cigar and genome location.
+     * </p>
+     * @param index the index of the target allele.
+     * @param paddingSize extra bases from the reference sequence to be added on either side.
+     * @param reference the reference to use as source.
+     * @return never {@code null}.
+     */
+    public Haplotype composeHaplotypeBasedOnReference(final int index, final int paddingSize, final ReferenceMultiSource reference)  {
+        Utils.nonNull(reference, "the input reference cannot be null");
+        ParamUtils.isPositiveOrZero(paddingSize, "the input padding must be 0 or greater");
+        ParamUtils.inRange(index, 0, 1, "the input allele index must be 0 or 1");
+
+        final SAMSequenceDictionary dictionary = reference.getReferenceSequenceDictionary(null);
+        if (dictionary == null) {
+            throw new IllegalArgumentException("the input reference does not have a dictionary");
+        }
+        final SAMSequenceRecord contigRecord = dictionary.getSequence(getContig());
+        if (contigRecord == null) {
+            throw new IllegalArgumentException("the input reference does not have a contig named: " + getContig());
+        }
+        final int contigLength = contigRecord.getSequenceLength();
+        if (contigLength < getEnd()) {
+            throw new IllegalArgumentException(String.format("this variant goes beyond the end of "
+                    + "the containing contig based on the input reference: "
+                    + "contig %s length is %d but variant end is %d", getContig(), contigLength, getEnd()));
+        }
+        final SimpleInterval referenceInterval = new SimpleInterval(getContig(),
+                Math.max(1, getStart() - paddingSize + 1),
+                Math.min(contigLength, getEnd() + paddingSize));
+
+        final ReferenceBases bases;
+        try {
+            bases = reference.getReferenceBases(null, referenceInterval);
+        } catch (final IOException ex) {
+            throw new GATKException("could not read reference file");
+        }
+        if (index == 0) {
```

Comment:
In think that given that this is a VC it is guaranteed that ref is at index 0. I imagine those problems where due to assume that the index 0 in order Allele containing objects also meant reference for example Allele array that was populated in a arbitrary manner... that said there is no much harm in calling getAllele(i) to verify, is just a added instructions.